src/components/onboarding/ChoosePlan.tsx

```diff
@@ -13,6 +17,8 @@ export const OnboardingChoosePlan: React.FC<OnboardingChoosePlanProps> = ({
 }) => {
   const trackEvent = useTrack();
   const mutation = useUpdateAccount();
+  const { data: account } = useAccount();
+  const canTrialGrowth = !account?.trialPeriodEndedAt;
```

Comment:
this does not take into account that `useAccount()` is async and may be loading, meaning while the account is loading `canTrialGrowth` will always be true

If that is the intended behavior that's fine but calling it out because it's not explicitly specified and rather a side-effect of the `!` + optional chaining here

Reply:
This should be safe because a user who signs up and is seeing this hasn't had a chance to claim the growth plan yet. The default in this case should be true I'd say

Also we have a server-side check to see if they're eligible for the free trial so if a bad actor attempts to abuse this we'll just charge them the full price (no free trial)

Good catch though because this is something that is pretty easy to miss🙏

Reply:
Actually I guess in this case, it should always be true

I'll update this to make leave it true and pass it through

Reply:
Done ✅